libs/data-service-generator/src/prepare-context.ts

```diff
@@ -209,11 +209,9 @@ function resolveLookupFields(entities: Entity[]): Entity[] {
             !fieldProperties.allowMultipleSelection &&
             !relatedFieldProperties.allowMultipleSelection;
 
-          //**@todo: in one-to-one relation, only one side should have a foreign key.
-          //We currently decide randomly based on sorting the permanent ID
-          //instead we should let the user decide which side holds the foreign key  */
           const isOneToOneWithoutForeignKey =
-            isOneToOne && field.permanentId > relatedField.permanentId;
+            (isOneToOne && field.permanentId !== fieldProperties.fkHolder) ||
+            (isOneToOne && field.permanentId > relatedField.permanentId);
```

Comment:
separate conditions for when fieldProperties.fkHolder is null (or undefined) and not null